src/boundary_integrals.jl

```diff
@@ -0,0 +1,150 @@
+##################
+# All 1D RefCube #
+##################
+function create_boundary_quad_rule{T}(::FunctionSpace{1, RefCube}, quad_rule::QuadratureRule{0, T})
+    w = weights(quad_rule)
+    boundary_quad_rule = QuadratureRule{1, T}[]
+
+    # Boundary 1
+    new_points = [-1.0]
+    push!(boundary_quad_rule, QuadratureRule(w,reinterpret(Vec{1, T}, new_points, (length(w),))))
+    # Boundary 2
+    new_points = [1.0]
+    push!(boundary_quad_rule, QuadratureRule(w,reinterpret(Vec{1, T}, new_points,(length(w),))))
+
+    return boundary_quad_rule
+end
+
+detJ_boundary{T}(::FunctionSpace{1, RefCube}, ::Tensor{2, 1, T}, ::Int) = one(T)
+
+##################
+# All 2D RefCube #
+##################
+function create_boundary_quad_rule{T}(::FunctionSpace{2, RefCube}, quad_rule::QuadratureRule{1, T})
+    w = weights(quad_rule)
+    p = reinterpret(T,points(quad_rule),(1,length(w)))
+    t = p[1,:]'
+    boundary_quad_rule = QuadratureRule{2, T}[]
+
+    # Boundary 1
+    new_points = [t; -ones(t)] # ξ = t, η = -1
+    push!(boundary_quad_rule, QuadratureRule(w,reinterpret(Vec{2, T},new_points,(length(w),))))
```

Comment:
For the `QuadratureRule` we need them in a vector anyway. As I see it the option is to do:

```jl
# Boundary 1
new_points = Vec{2,T}[]
for i in 1:length(w)
    push!(new_points, Vec{2,T}((t[i], -one(T))))
end
push!(boundary_quad_rule, QuadratureRule(w, new_points))
# Boundary 2
...
```

or perhaps
```jl
# Boundary 1
new_points = Array(Vec{2,T}, length(w))
for i in 1:length(w)
    new_points[i] = Vec{2,T}((t[i], -one(T)))
end
push!(boundary_quad_rule, QuadratureRule(w, new_points))
# Boundary 2
...
```
Don't know whats the most efficient. Could probably use @ inbounds aswell